WordPress/Sniffs/WP/CapabilitiesSniff.php

```diff
@@ -0,0 +1,470 @@
+		// If the parameter is anything other than T_CONSTANT_ENCAPSED_STRING throw a warning and bow out.
+		$first_non_empty = null;
+		for ( $i = $parameter['start']; $i <= $parameter['end']; $i++ ) {
+			if ( isset( Tokens::$emptyTokens[ $this->tokens[ $i ]['code'] ] ) ) {
+				continue;
+			}
+
+			if ( \T_CONSTANT_ENCAPSED_STRING !== $this->tokens[ $i ]['code']
+				|| null !== $first_non_empty
+			) {
+				// Throw warning at low severity.
+				$this->phpcsFile->addWarning(
+					"Couldn't determine the value passed to the capability parameter. Please check if it matches a valid capability.",
+					$i,
+					'NeedsManualCheck',
+					array( $parameter['clean'] ),
+					3 // Message severity set to below default.
+				);
+				return;
+			}
+
+			$first_non_empty = $i;
+		}
+
+		if ( null === $first_non_empty ) {
+			// Parse error. Bow out.
+			return;
+		}
+
+		/*
+		 * As of this point we know that the `$capabilities` parameter only contains the one token
+		 * and that that token is a `T_CONSTANT_ENCAPSED_STRING`.
+		 */
+		$matched_parameter = TextStrings::stripQuotes( $this->tokens[ $first_non_empty ]['content'] );
+
+		if ( isset( $this->core_capabilities[ $matched_parameter ] ) ) {
+			return;
+		}
+
+		if ( empty( $matched_parameter ) ) {
+			$this->phpcsFile->addError(
+				'Found an empty capability parameter in the function call "%s()"',
+				$first_non_empty,
+				'Invalid',
+				array( $matched_content )
+			);
+			return;
+		}
+
+		// Check if additional capabilities were registered via the ruleset and if the found capability matches any of those.
+		$custom_capabilities = $this->merge_custom_array( $this->custom_capabilities, array() );
+		if ( isset( $custom_capabilities[ $matched_parameter ] ) ) {
+			return;
+		}
+
+		if ( isset( $this->deprecated_capabilities[ $matched_parameter ] ) ) {
+			$this->get_wp_version_from_cli( $this->phpcsFile );
+			$is_error = version_compare( $this->deprecated_capabilities[ $matched_parameter ], $this->minimum_supported_version, '<' );
+
+			$data = array(
+				$matched_parameter,
+				$matched_content,
+				$this->deprecated_capabilities[ $matched_parameter ],
+			);
+
+			MessageHelper::addMessage(
+				$this->phpcsFile,
+				'The capability "%s", found in the function call "%s()", has been deprecated since WordPress version %s.',
+				$first_non_empty,
+				$is_error,
+				'Deprecated',
+				$data
+			);
+			return;
+		}
+
+		if ( isset( $this->core_roles[ $matched_parameter ] ) ) {
+			$this->phpcsFile->addError(
+				'Capabilities should be used instead of roles. Found "%s" in a function call "%s()"',
+				$first_non_empty,
+				'RoleFound',
+				array(
+					$matched_parameter,
+					$matched_content,
+				)
+			);
+			return;
+		}
+
+		$this->phpcsFile->addWarning(
+			'"%s" is an unknown capability. Check the "%s()" function call, and that the capability is registered with WordPress. If this is a custom capability, make sure you have passed it to the custom_capabilities sniff property.',
```

Comment:
```suggestion
			'Found unknown capability "%s" in function call to %s(). Please check the spelling of the capability. If this is a custom capability, please verify the capability is registered with WordPress via a call to WP_Role(s)->add_cap().' . \PHP_EOL . 'Custom capabilities can be made known to this sniff by setting the "custom_capabilities" property in the PHPCS ruleset.',
```

Along the same lines as above. Trying to make the error message more actionable for end-users. Hoping this makes it clearer ?